src/World.cc

```diff
@@ -245,48 +250,51 @@ Errors World::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
 
   for (const auto &ifaceModelPair : this->dataPtr->interfaceModels)
   {
-    frameNames.insert(ifaceModelPair.second->Name());
+    implicitFrameNames.insert(ifaceModelPair.second->Name());
   }
 
-  // Load all the physics.
-  if (_sdf->HasElement("physics"))
+  for (auto elem = _sdf->GetFirstElement(); elem; elem = elem->GetNextElement())
   {
-    this->dataPtr->physics.clear();
-    Errors physicsLoadErrors = loadUniqueRepeated<Physics>(_sdf, "physics",
-        this->dataPtr->physics);
-    errors.insert(errors.end(), physicsLoadErrors.begin(),
-        physicsLoadErrors.end());
+    const std::string elementName = elem->GetName();
+    if (elementName == "model")
+    {
+      auto model = loadSingle<Model>(errors, elem, _config);
+      if (!recordUniqueName(nestedModelNames, elementName, model.Name()))
+      {
+        continue;
+      }
+      implicitFrameNames.insert(model.Name());
+      this->dataPtr->models.push_back(std::move(model));
+    }
+    else if (elementName == "joint")
+    {
+      auto joint = loadSingle<Joint>(errors, elem);
+      if (!recordUniqueName(jointNames, elementName, joint.Name()))
+      {
+        continue;
+      }
```

Comment:
I think you should add `implicitFrameNames.insert(joint.Name());`

Reply:
Okay, but this will also be a change in behavior. We weren't doing that before, but since world joints are new, this would be a reasonable bug fix.

Reply:
ok, we can rely on the frame graph errors then